astro/src/content/docs/extend/code/lambdas/testing.mdx

```diff
@@ -0,0 +1,429 @@
+</Aside>
+
+## Manually Create A Simple Lambda
+
+Let's start by making a simple lambda to test that it works on your machine.
+
+- Log in to the [FusionAuth admin UI](http://localhost:9011/admin).
+- Navigate to "Customizations -> Lambdas".
+- Click the <IconButton icon="plus" color="green" /> button at the top right to add a new lambda.
+- Enter the Id `f3b3b547-7754-452d-8729-21b50d111505`.
+- Enter the Name `[ATest]` (to put it at the top of the list of lambdas alphabetically).
+- Select the Type `JWT Populate`.
+- Leave the Engine as `GraalJS`.
+- Enable <InlineUIElement>Debug Enabled</InlineUIElement> so that you can see messages in the event log.
+- Add the line `jwt.message = 'Hello World!';` to the body of the `populate` function.
+
+The body should now be similar to below.
+
+```js
+function populate(jwt, user, registration) {
+  jwt.message = 'Hello World!';
+  console.info('Hello World!');
+}
+```
+
+<img src="/img/docs/extend/code/lambdas/populate-lambda.png" alt="Creating a JWT populate lambda" role="bottom-cropped" width="1200" />
+
+Save the lambda.
+
+Now activate the lambda for the example app.
+
+- Navigate to "Applications" and click the <IconButton icon="edit" color="blue" /> button on the "Example app".
+- Click on the "JWT" tab.
+- Toggle <InlineUIElement>Enabled</InlineUIElement> to on.
+- Under "Lambda Settings", select the lambda you created, called "[ATest]", for the `Access Token populate lambda`.
+- Click the <IconButton icon="save" color="blue" /> button to save the changes.
+
+<img src="/img/docs/extend/code/lambdas/edit-application.png" alt="Enabling the lambda in an application" className="img-fluid" width="1200" />
+
+You can now test that the new lambda writes to the event log and returns extra data in the JWT. The repository you downloaded contains two directories.
+- `complete-application` — This is the result of the work you will complete in this guide if you need to refer to the finished files. Do not work in this directory.
+- `app` — This contains a basic application to which you will add tests in this guide. Work in this directory.
+
+Open the `routes/index.js` file in the `app` directory.
+
+Add a line in the OAuth redirect route handler to write the JWT to the console at login. The start of the function should now look like the code below.
+
+```js
+router.get('/oauth-redirect', function (req, res, next) {
+  console.dir(res);
+```
+
+Save the file and start the test app with the following command.
+
+```js
+npm start
+```
+
+Log in to the app at `http://localhost:3000` with user `richard@example.com` and password `password`.
+
+In the FusionAuth admin UI, navigate to "System -> Event Log" and you will see a log entry of the invocation of your "Hello World" lambda. The entire HTTP response is logged in the test app terminal. The JWT is a long alphanumeric line at the end of the response. It should look something like below.
```

Comment:
I am getting an error here. `Unexpected error while decrypting a [fusionauth.known-device] cookie.

Exception:
javax.crypto.BadPaddingException: Given final block not properly padded. Such issues can arise if a bad key is used during decryption.
	at java.base/com.sun.crypto.provider.CipherCore.unpad(CipherCore.java:862)
	at java.base/com.sun.crypto.provider.CipherCore.fillOutputBuffer(CipherCore.java:942)
	at java.base/com.sun.crypto.provider.CipherCore.doFinal(CipherCore.java:826)
	at java.base/com.sun.crypto.provider.AESCipher.engineDoFinal(AESCipher.java:480)
	at java.base/javax.crypto.Cipher.doFinal(Cipher.java:2152).....
`

Can't get much further..